core/src/main/java/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java

```diff
@@ -582,4 +584,23 @@ private Answer execute(AggregationControlCommand cmd) {
         }
         return new Answer(cmd, false, "Fail to recognize aggregation action " + action.toString());
     }
+
+    public boolean isSystemVMSetup(String vmName, String controlIp) throws InterruptedException {
+        if (vmName.startsWith("s-") || vmName.startsWith("v-")) {
+            ScriptConfigItem scriptConfigItem = new ScriptConfigItem(VRScripts.SYSTEM_VM_PATCHED, "/opt/cloud/bin/keystore*");
+            ExecutionResult result = new ExecutionResult(false, "");
+            int retries = 0;
+            while (!result.isSuccess() && retries < 600) {
```

Comment:
Not saying its wrong but what's the logic behind the value 600? Should it be parametrized?

Reply:
the idea was to wait for 10mins to allow systemVMs to patch

Reply:
by parameterised do you mean a global setting @nvazquez ?

Reply:
Thanks, yes that was my idea, could be useful but can live without it :)